edge-util/src/Microsoft.Azure.Devices.Edge.Util/Option.cs

```diff
@@ -16,6 +16,8 @@ public struct Option<T> : IEquatable<Option<T>>
 
         internal Option(T value, bool hasValue)
         {
+            Preconditions.CheckArgument(!hasValue || value != null , "Value cannot be null if hasValue flag is set to true.");
```

Comment:
I usually prefer having Preconditions checks on public methods, so that you get immediate context on what is going on. Since nobody using this type ever sets the hasValue flag, the message can be a bit confusing.
I would recommend adding  CheckNotNull check to Some(T value) method instead. That will also throw a ArgumentNullException like we want.

Reply:
Sure, will change. Good point!